producer/writer/router.go

```diff
@@ -20,7 +20,54 @@
 
 package writer
 
+import (
+	"fmt"
+	"sync"
+)
+
 type ackRouter interface {
 	// Ack acks the metadata.
 	Ack(ack metadata) error
+
+	// Register registers a message writer.
+	Register(replicatedShardID uint64, mw messageWriter)
+
+	// Unregister removes a message writer.
+	Unregister(replicatedShardID uint64)
+}
+
+type router struct {
+	sync.RWMutex
+
+	mws map[uint64]messageWriter
```

Comment:
nit: It's a little unclear what mws stands for - message writer shards?

Reply:
will rename to `messageWriters` to be more clear